icu4c/source/common/unicode/stringpiece.h

```diff
@@ -130,13 +130,13 @@ class U_COMMON_API StringPiece : public UMemory {
    * @stable ICU 65
    */
   template <typename T,
-            typename = typename std::enable_if<
-                (std::is_same<decltype(T().data()), const char*>::value
+            typename = typename std::enable_if_t<
```

Comment:
```suggestion
            typename = std::enable_if_t<
```

That `typename` was needed because of the `::type` but should no longer be needed.